python/ray/exceptions.py

```diff
@@ -50,6 +50,13 @@ def from_ray_exception(ray_exception):
                 return pickle.loads(ray_exception.serialized_exception)
             except Exception as e:
                 msg = "Failed to unpickle serialized exception"
+                # Include a fallback string/stacktrace to aid debugging.
+                formatted = getattr(
+                    ray_exception,
+                    "formatted_exception_string",
```

Comment:
where does `formatted_exception_string` come from? please drop a comment here explaining, as it looks like black magic to an uninformed reader